src/components/vm/nics/nicBody.tsx

```diff
@@ -271,3 +302,355 @@ export const NetworkTypeAndSourceRow = ({
         </>
     );
 };
+
+export function portForwardText(pf: VMInterfacePortForward): string {
+    let text = "";
+
+    if (pf.address)
+        text += pf.address + ":";
+    if (pf.dev)
+        text += pf.dev + ":";
+    if (!pf.range.some(r => r.exclude != "yes"))
+        text += _("all");
+    for (let i = 0; i < pf.range.length; i++) {
+        const r = pf.range[i];
+        if (i > 0)
+            text += ",";
+        if (r.exclude == "yes")
+            text += _("excluding ");
+        text += r.start;
+        if (r.end)
+            text += "-" + r.end;
+        text += " → " + (r.to || r.start) + (r.end ? "-" + (Number(r.to || r.start) + Number(r.end) - Number(r.start)).toString() : "");
+        text += "/" + pf.proto;
```

Comment:
These 17 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2331-c59dd4a9-20251031-071718-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)